GVRf/Framework/framework/src/main/jni/objects/components/render_data.h

```diff
@@ -245,18 +328,44 @@ class RenderData: public Component {
 
     void set_draw_mode(GLenum draw_mode) {
         draw_mode_ = draw_mode;
+        hash_code_dirty_ = true;
     }
 
     void set_texture_capturer(TextureCapturer *capturer) {
         texture_capturer = capturer;
     }
-
+    // TODO: need to consider texture_capturer in hash_code ?
     TextureCapturer *get_texture_capturer() {
         return texture_capturer;
     }
 
+    std::string getHashCode(){
+    	if(hash_code_dirty_){
+    		std::string render_data_string;
+
+    		render_data_string.append(to_string(use_light_));
+    		render_data_string.append(to_string(light_));
+    		render_data_string.append(to_string(getComponentType()));
+    		render_data_string.append(to_string(use_lightmap_));
+    		render_data_string.append(to_string(render_mask_));
+    		render_data_string.append(to_string(offset_));
+    		render_data_string.append(to_string(offset_factor_));
+    		render_data_string.append(to_string(offset_units_));
+    		render_data_string.append(to_string(depth_test_));
+    		render_data_string.append(to_string(alpha_blend_));
+    		render_data_string.append(to_string(alpha_to_coverage_));
+    		render_data_string.append(to_string(sample_coverage_));
+    		render_data_string.append(to_string(invert_coverage_mask_));
+    		render_data_string.append(to_string(draw_mode_));
+
+    		hash_code = render_data_string;
+    		hash_code_dirty_ = false;
+
+    	}
+    	return hash_code;
```

Comment:
Please use a local string for creating the hash code. There shouldn't be a data member called hash_code. It is also not cleared. C++ will optimize on the return value. So there is no performance concern.